diff --git a/ios/Sources/Extensions+Helpers/RCTView+Helpers.swift b/ios/Sources/Extensions+Helpers/RCTView+Helpers.swift
index 09be306d5aa39337c5114c2ad6ba7513218e0751..d5ab2f2df62f85b9bd6a65ae98a975bf98b97a96 100644
--- a/ios/Sources/Extensions+Helpers/RCTView+Helpers.swift
+++ b/ios/Sources/Extensions+Helpers/RCTView+Helpers.swift
@@ -25,15 +25,8 @@ public extension RCTView {
     return rootView.recursivelyFindSubview(whereType: targetType);
   };
   
-  var closestParentReactContentView: RCTRootContentView? {
-    let targetType = RCTRootContentView.self;
-  
-    if let match = self.recursivelyFindParentView(whereType: targetType) {
-      return match;
-    };
-    
-    guard let rootView = self.rootViewForCurrentWindow else { return nil };
-    return rootView.recursivelyFindSubview(whereType: targetType);
+  var closestParentReactContentView: RCTRootView? {
+    return closestParentRootView;
   };
   
   var reactTouchHandlers: [RCTTouchHandler]? {
@@ -58,6 +51,8 @@ public extension RCTView {
       currentView = superview;
     };
     
-    return closestParentReactContentView?.reactTouchHandlers?.first;
+    return closestParentReactContentView?.gestureRecognizers?
+      .compactMap { $0 as? RCTTouchHandler }
+      .first;
   };
 };
